aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -364,8 +392,18 @@ inline Return Dispatcher::callWithDispatchKey(const TypedOperatorHandle<Return(A
   detail::unused_arg_(args...);  // workaround for a false-positive warning about unused parameters in gcc 5
   // No alias dispatch key is allowed at runtime.
   TORCH_INTERNAL_ASSERT_DEBUG_ONLY(!c10::isAliasDispatchKey(dispatchKey));
+  auto dispatchKeySet = op.operatorIterator_->op.dispatchKeyExtractor()
```

Comment:
I still haven't done this- I'll try this in a bit. One thing to note is that this function is no longer used directly in the standard entry point to the dispatcher (the C++ API), so it will only affect programs that call it directly. After this PR it's only called by tests in PyTorch core.

That means that I don't know of any benchmarks that test out this function currently. I'm thinking of testing it by temporarily changing the codegen'd autograd kernels to redispatch to their CPU kernels with `Dispatcher::callWithDispatchKey(..CPU..)` , and comparing that to master (where the autograd kernels currently use `Dispatcher::call(...)`. But let me know if you don't like that benchmark / have an improvement in mind.